database/src/main/dataset_schema/list_schemas.sql

```diff
@@ -0,0 +1,50 @@
+/*
+ * Copyright 2018 ABSA Group Limited
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+DROP FUNCTION IF EXISTS dataset_schema.list_schemas(BOOLEAN);
+
+CREATE OR REPLACE FUNCTION dataset_schema.list_schemas(
+    IN  i_include_disabled      BOOLEAN DEFAULT FALSE,
+    OUT schema_name             TEXT,
+    OUT schema_latest_version   INTEGER,
+    OUT disabled                BOOLEAN
+) RETURNS SETOF record AS
+$$
+-------------------------------------------------------------------------------
+--
+-- Function: jobs_configuration.list_schemas(1)
+--      Returns a list of schemas with their latest versions
+--
+-- Parameters:
+--      i_include_disabled       - flag indicating if to include disabled schemas too
+--
+-- Returns:
+--      schema_name             - name of the schema
+--      schema_latest_version   - the latest version of the schema
+--
+-------------------------------------------------------------------------------
+DECLARE
+BEGIN
+    RETURN QUERY
+    SELECT dsh.schema_name, dsh.schema_latest_version, dsh.disabled_when IS NOT NULL
+    FROM dataset_schema.heads dsh
+    WHERE i_include_disabled OR dsh.disabled_when IS NULL
+    ORDER BY schema_name; --TODO Include order by?
```

Comment:
Current implementation is ordered by version

Reply:
This is a list of schemas. It doesn't make sense to have it ordered by the schema version when only the latest version number is returned. But reminds me, that a function returning the history  (all versions) of a schema will probably be needed.

Reply:
🤦‍♂️ Sorry, name, it is ordered by name.

Reply:
> This is a list of schemas. It doesn't make sense to have it ordered by the schema version when only the latest version number is returned. But reminds me, that a function returning the history (all versions) of a schema will probably be needed.

I presume these are not all the functions. If they are supposed to be, I will supply the list I come up with